hpke_test.go

```diff
@@ -155,3 +159,347 @@ func TestModes(t *testing.T) {
 		}
 	}
 }
+
+func processTestVectorEncryptions(t *testing.T, vector testVector, enc *EncryptContext, dec *DecryptContext) {
+	for _, data := range vector.Encryptions {
+		dataAad, err := hex.DecodeString(data.Aad)
+		if err != nil {
+			t.Fatalf("[%x, %x, %x] Error in DecodeString: %s", vector.KEMID, vector.KDFID, vector.AEADID, err)
+		}
+
+		dataPlaintext, err := hex.DecodeString(data.Plaintext)
+		if err != nil {
+			t.Fatalf("[%x, %x, %x] Error in DecodeString: %s", vector.KEMID, vector.KDFID, vector.AEADID, err)
+		}
+
+		dataCiphertext, err := hex.DecodeString(data.Ciphertext)
+		if err != nil {
+			t.Fatalf("[%x, %x, %x] Error in DecodeString: %s", vector.KEMID, vector.KDFID, vector.AEADID, err)
+		}
+
+		encrypted := enc.Seal(dataAad, dataPlaintext)
+		decrypted, err := dec.Open(dataAad, encrypted)
+		if err != nil {
+			t.Fatalf("[%x, %x, %x] Error in Open: %s", vector.KEMID, vector.KDFID, vector.AEADID, err)
+		}
+
+		if dataCiphertext != nil && !bytes.Equal(encrypted, dataCiphertext) {
+			t.Fatalf("[%x, %x, %x] Incorrect encryption: [%x] != [%x]", vector.KEMID, vector.KDFID, vector.AEADID, dataCiphertext, encrypted)
+		}
+
+		if !bytes.Equal(decrypted, dataPlaintext) {
+			t.Fatalf("[%x, %x, %x] Incorrect decryption: [%x] != [%x]", vector.KEMID, vector.KDFID, vector.AEADID, decrypted, dataPlaintext)
+		}
+	}
+}
+
+func processTestVector(t *testing.T, vectorString string) {
```

Comment:
It seems like you should have initiator and responder sides to this.  The Initiator side will call `SetupXI` and the responder side will call `SetupXR`.

In order to support that, I expect you'll want to pull out all the unmarshal logic below into `MarshalJSON` and `UnmarshalJSON` functions for the `testVector` struct, and have the fields in that struct actually just be the real `KEMPrivateKey`, `KEMPublicKey`, `[]byte`, etc. values.